ARCHITECTURE.md

```diff
@@ -2,19 +2,54 @@ This document aims to outline at a high level the various parts that make up map
 
-Code and build scripts belonging to platform SDKs are contained in the `platform` directory, which has subdirectories for each platform. The `platform/darwin` and `platform/default` directories contain code shared by multiple platform SDKs.
+## Platform specific code
+
+Code and build scripts belonging to platform SDKs are contained in the `platform` directory, which has subdirectories for each platform. 
+
+- `platform/darwin` and `platform/default` directories contain code shared by multiple platform SDKs.
+
+- `platform/ios` and `platform/macos` - the SDKS for Apple's operating systems, https://github.com/mapbox/mapbox-gl-native-ios/commit/a139216 (mapbox hosted both iOS and MacOS SDKs in the same project).
+
+- `platform/android` - Android SDK, forked from https://github.com/mapbox/mapbox-gl-native-android/commit/4c12fb2c.
+- `platform/glfw` - [GLFW](https://www.glfw.org) is library to support OpenGL development on the desktop. The code in this directory builds an executable application for demo/dev/local testing purposes (e.g. https://github.com/mapbox/mapbox-gl-native/blob/master/README.md?plain=1#L27).
+
+## Maplibre-gl-js
+
+`maplibre-gl-js` is added to this repostiory as a top-level submodule to provided
+
+- Test cases and test data, e.g. all rendering test manifests json files under `render-test` load test cases from `maplibre-gl-js/test/*`.
+- Shader written in GLSL ES.
+
+## Other directories
+
+- `benchmark` contains the performance tests built using https://github.com/google/benchmark/. The code under this directory builds the ``mbgl-benchmark-test` executable to execute the benchmark tests.
+  - The entry point of this executable is `platform/default/src/mbgl/benchmark/main.cpp -> benchmark/src/mbgl/benchmark/benchmark.cpp`.
+  - iOS SDK rans benchmark test through a separate app named `BenchmarkApp`.
+  - Android SDK does not have benchmark test.
+
+- `bin` contains the code for tools like `mbgl-cache`, `mbgl-offline`, and `mbgl-render`.
+- `expression-test` contains tests for the expression feature in the map style (see more details about expression [here](https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/) and [here](https://github.com/mapbox/mapbox-gl-native/wiki/Expression-Architecture)).
+- `metrics` contains test manifest files and ground truth for graphic comparison based render test.
+- `misc` contains protobuf for style, vector tile, and glyphs. It also icons and pictures used in documents.
+- `render-test` contains image diff based render tests. These tests verify if the rendering results match with expectations by capturing the rendering results and compare with the groundtruth images in the `metrics` directory.
+  - A typical render test run can be triggered by `mbgl-render-test-runner --manifestPath metrics/linux-clang8-release-style.json` . It will by default generate a html file `linux-clang8-release-style.html` with test results visualized and summarized.
```

Comment:
```suggestion
  - A typical render test run can be triggered by `mbgl-render-test-runner --manifestPath metrics/linux-clang8-release-style.json`. It will by default generate a html file `linux-clang8-release-style.html` with test results visualized and summarized.
```